File: solutions/removeElement-27/removeElement-27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:       
        n = len(nums)
        if n==0:
            return 0
        left = 0
        right = 0
        
        while right<n:
            
            if nums[right] != val:
                nums[left] = nums[right]
                left += 1
            right += 1
        
        return left

File: solutions/removeElement-27/removeElement-27.cpp

/**********************************************************************************
*
* Given an array nums and a value val, remove all instances of that value in-place and return the new length.
*
* Input: nums = [3,2,2,3], val = 3
* Output: 2, nums = [2,2]
*
*
**********************************************************************************/

class Solution {

  /*
  *   [Two Pointers]
  *
  *   Time complexity : O(n). Assume the array has a total of n elements, both i and j traverse at most 2n steps.
  *
  *   Space Complexity: O(1).
  */

public:
    int removeElement(vector<int>& nums, int val) {
        
        int n = nums.size();
        
        int slow = 0;
        int fast = 0;
        
        while (fast < n) {
            
            if(nums[fast] != val) {
                
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }
        return slow;
    }
};